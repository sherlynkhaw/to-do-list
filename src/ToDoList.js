import React, {Component} from 'react';
import {withStyles} from 'material-ui/styles';
import Card, {CardActions, CardContent} from 'material-ui/Card';
import List, {ListItem, ListItemSecondaryAction, ListItemText} from 'material-ui/List';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';
import MyList from './List';

const styles = theme => ({
    card: {
        width: 275,
        margin: 20,
        padding: 30
    },
});

class ToDoList extends Component {

    constructor() {
        super();
        this.list = new MyList();

        this.state = {
            list: this.list.items,
            newItem: ""
        };
    }

    handleChange(event) {
        this.setState({
            newItem: event.target.value
        });
    }

    handleAdd() {
        this.list.add(this.state.newItem);

        this.setState({
            list: this.list.items,
            newItem: ""
        });
    }

    render() {
        const { classes } = this.props;

        return (
            <div>
                <Card className={classes.card}>
                    <CardContent>
                        {/*List Items*/}
                        <List>
                            {this.state.list.map((item, index) => (
                                <ListItem
                                    key={index}
                                >
                                    <ListItemText primary={item}/>
                                    <ListItemSecondaryAction>
                                    </ListItemSecondaryAction>
                                </ListItem>
                            ))}
                        </List>
                    </CardContent>
                    <CardActions>
                        {/*List Input*/}
                        <TextField
                            id="listNewItem"
                            label="New Item"
                            margin="normal"
                            value={this.state.newItem}
                            onChange={this.handleChange.bind(this)}
                        />
                        <Button
                            id="addButton"
                            raised
                            onClick={this.handleAdd.bind(this)}
                        >
                            Add
                        </Button>
                    </CardActions>
                </Card>
            </div>
        );
    }
}

export default withStyles(styles)(ToDoList);