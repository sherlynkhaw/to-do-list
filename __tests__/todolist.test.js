import React from 'react';
import ToDoList from '../src/ToDoList';
import renderer from 'react-test-renderer';

test('Adding New Item to the List', () => {
    const component = renderer.create(
        <ToDoList/>
    );
    const root = component.root;
    const instance = root.instance;

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // Manually add "Hello" to state
    instance.setState({ newItem: "Hello" });
    expect(tree).toMatchSnapshot();

    // Button
    const button = root.findByProps({id: "addButton"});
    button.props.onClick();
    expect(tree).toMatchSnapshot();
});