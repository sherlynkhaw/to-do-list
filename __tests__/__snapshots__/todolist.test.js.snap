// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adding New Item to the List 1`] = `
<div>
  <div
    className="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 ToDoList-card-1"
  >
    <div
      className="MuiCardContent-root-29"
    >
      <ul
        className="MuiList-root-30 MuiList-padding-31"
      />
    </div>
    <div
      className="MuiCardActions-root-34"
    >
      <div
        className="MuiFormControl-root-36 MuiFormControl-marginNormal-37 MuiCardActions-action-35"
        onBlur={[Function]}
        onFocus={[Function]}
        required={false}
      >
        <label
          className="MuiFormLabel-root-46 MuiInputLabel-root-40 MuiInputLabel-formControl-41 MuiInputLabel-animated-44"
          data-shrink={false}
          htmlFor="listNewItem"
        >
          New Item
        </label>
        <div
          className="MuiInput-root-50 MuiInput-formControl-51 MuiInput-inkbar-52 MuiInput-underline-56"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-59 MuiInput-inputSingleline-62"
            defaultValue={undefined}
            disabled={false}
            id="listNewItem"
            name={undefined}
            onChange={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-80 MuiButton-root-65 MuiButton-raised-72 MuiCardActions-action-35"
        disabled={false}
        id="addButton"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label-67"
        >
          Add
        </span>
        <span
          className="MuiTouchRipple-root-82"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Adding New Item to the List 2`] = `
<div>
  <div
    className="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 ToDoList-card-1"
  >
    <div
      className="MuiCardContent-root-29"
    >
      <ul
        className="MuiList-root-30 MuiList-padding-31"
      />
    </div>
    <div
      className="MuiCardActions-root-34"
    >
      <div
        className="MuiFormControl-root-36 MuiFormControl-marginNormal-37 MuiCardActions-action-35"
        onBlur={[Function]}
        onFocus={[Function]}
        required={false}
      >
        <label
          className="MuiFormLabel-root-46 MuiInputLabel-root-40 MuiInputLabel-formControl-41 MuiInputLabel-animated-44"
          data-shrink={false}
          htmlFor="listNewItem"
        >
          New Item
        </label>
        <div
          className="MuiInput-root-50 MuiInput-formControl-51 MuiInput-inkbar-52 MuiInput-underline-56"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-59 MuiInput-inputSingleline-62"
            defaultValue={undefined}
            disabled={false}
            id="listNewItem"
            name={undefined}
            onChange={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-80 MuiButton-root-65 MuiButton-raised-72 MuiCardActions-action-35"
        disabled={false}
        id="addButton"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label-67"
        >
          Add
        </span>
        <span
          className="MuiTouchRipple-root-82"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Adding New Item to the List 3`] = `
<div>
  <div
    className="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 ToDoList-card-1"
  >
    <div
      className="MuiCardContent-root-29"
    >
      <ul
        className="MuiList-root-30 MuiList-padding-31"
      />
    </div>
    <div
      className="MuiCardActions-root-34"
    >
      <div
        className="MuiFormControl-root-36 MuiFormControl-marginNormal-37 MuiCardActions-action-35"
        onBlur={[Function]}
        onFocus={[Function]}
        required={false}
      >
        <label
          className="MuiFormLabel-root-46 MuiInputLabel-root-40 MuiInputLabel-formControl-41 MuiInputLabel-animated-44"
          data-shrink={false}
          htmlFor="listNewItem"
        >
          New Item
        </label>
        <div
          className="MuiInput-root-50 MuiInput-formControl-51 MuiInput-inkbar-52 MuiInput-underline-56"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-59 MuiInput-inputSingleline-62"
            defaultValue={undefined}
            disabled={false}
            id="listNewItem"
            name={undefined}
            onChange={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-80 MuiButton-root-65 MuiButton-raised-72 MuiCardActions-action-35"
        disabled={false}
        id="addButton"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label-67"
        >
          Add
        </span>
        <span
          className="MuiTouchRipple-root-82"
        />
      </button>
    </div>
  </div>
</div>
`;
